### PR TITLE
Upgrade to PixiJS v6

### PR DESCRIPTION
**Pull request by bigtimebuddy**
## Changes

* Upgrade to PixiJS v6 with package-level types
* Upgrade to npm 7
* Upgrade Rollup
* Replace JSDoc with Webdoc
* Upgrade Floss and Electron
* Use API-Extractor to generate types
* Replace tslint with eslint
* Convert tests to TypeScript
* Changed package name from `pixi-sound` to `@pixi/sound`

### CODE REVIEW

File: src/Sound.ts [hunk 1]
```diff
@@ -111,112 +103,91 @@ export class Sound
...
      *        with a `start` and `end`, which are the times in seconds. Optionally, can include
      *        a `speed` amount where 1 is 100% speed.
-     * @param {PIXI.sound.Sound~completeCallback} [options.complete=null] Global complete callback
+     * @param {Sound~completeCallback} [options.complete=null] - Global complete callback
      *        when play is finished.
-     * @param {PIXI.sound.Sound~loadedCallback} [options.loaded=null] Call when finished loading.
-     * @param {boolean} [options.loop=false] true to loop the audio playback.
-     * @return {PIXI.sound.Sound} Created sound instance.
+     * @param {Sound~loadedCallback} [options.loaded=null] - Call when finished loading.
+     * @param {boolean} [options.loop=false] - true to loop the audio playback.
```
Comment (andrewstart) [inline src/Sound.ts:191]: Would it be possible to move the options documentation to the Options interface (and drop the types/names)?

File: src/Sound.ts [hunk 2]
```diff
@@ -111,112 +103,91 @@ export class Sound
...
+     * @return {Sound} Created sound instance.
```
Comment (andrewstart) [inline src/Sound.ts:192]: Does Webdoc require the return type, or can we drop `{Sound}`?

File: src/Sound.ts [hunk 3]
```diff
@@ -510,34 +449,36 @@ export class Sound
...
+     * @param {Sound~completeCallback|object} options - Either completed function or play options.
+     * @param {number} [options.start=0] - Time when to play the sound in seconds.
+     * @param {number} [options.end] - Time to end playing in seconds.
+     * @param {String} [options.sprite] - Play a named sprite. Will override end, start and speed options.
+     * @param {number} [options.speed] - Override default speed, default to the Sound's speed setting.
+     * @param {number} [options.volume] - Current volume amount for instance.
+     * @param {boolean} [options.muted] - Override default muted, default to the Sound's muted setting.
+     * @param {boolean} [options.loop] - Override default loop, default to the Sound's loop setting.
+     * @param {Sound~completeCallback} [options.complete] - Callback when complete.
+     * @param {Sound~loadedCallback} [options.loaded] - If the sound isn't already preloaded, callback when
```
Comment (andrewstart) [inline src/Sound.ts:489]: As above, can these be moved to the interface itself?

Comment (bigtimebuddy) [reply src/Sound.ts:489]: Yes

File: src/Sound.ts (hunk 2 above)
Comment (bigtimebuddy) [reply src/Sound.ts:192]: We can drop it

File: src/Sound.ts (hunk 1 above)
Comment (bigtimebuddy) [reply src/Sound.ts:191]: Yup, good call

**Pull request merged**